buildstockbatch/gcp/gcp.py

```diff
@@ -0,0 +1,1267 @@
+            memory = pp_env.get("memory_mib", GcpBatch.DEFAULT_PP_MEMORY_MIB)
+
+            # Allowable values are documented at:
+            # https://cloud.google.com/python/docs/reference/run/latest/google.cloud.run_v2.types.ResourceRequirements
+            cpus_to_memory_limits = {
+                1: (512, 4096),
+                2: (512, 8192),
+                4: (2048, 16384),
+                8: (4096, 32768),
+            }
+
+            assert cpus in cpus_to_memory_limits, "gcp.postprocessing_environment.cpus must be 1, 2, 4 or 8."
+            min_memory, max_memory = cpus_to_memory_limits[cpus]
+            assert memory >= min_memory, (
+                f"gcp.postprocessing_environment.memory_mib must be at least {min_memory} for {cpus} CPUs. "
+                f"(Found {memory}) See https://cloud.google.com/run/docs/configuring/services/cpu"
+            )
+            assert memory <= max_memory, (
+                f"gcp.postprocessing_environment.memory_mib must be less than or equal to {max_memory} for {cpus} CPUs "
+                f"(Found {memory}) See https://cloud.google.com/run/docs/configuring/services/memory-limits"
+            )
+
+    @staticmethod
+    def validate_project(project_file):
+        super(GcpBatch, GcpBatch).validate_project(project_file)
+        GcpBatch.validate_gcp_args(project_file)
+
+    @property
+    def docker_image(self):
+        return "nrel/buildstockbatch"
+
+    @property
+    def results_dir(self):
+        return f"{self.gcs_bucket}/{self.gcs_prefix}/results"
+
+    @property
+    def registry_url(self):
+        """
+        The registry that the image(s) will be pushed to.
+
+        :returns: A string of a GCP Artifact Repository URL; for example,
+            `https://us-central1-docker.pkg.dev
+        """
+        return f"https://{self.repository_uri.split('/')[0]}"
+
+    @property
+    def repository_uri(self):
+        """
+        The "repository" (name) for this image for pushing to a GCP Artifact
+        Repository.
+
+        :returns: A string for this image given the Artifact Repository (given
+            its region, project name, and repo name), followed by
+            "buildstockbatch"; for example,
+             `us-central1-docker.pkg.dev/buildstockbatch/buildstockbatch-docker/buildstockbatch`
+        """
+        return f"{self.region}-docker.pkg.dev/{self.gcp_project}/{self.ar_repo}/buildstockbatch"
+
+    @property
+    def postprocessing_job_id(self):
+        return f"{self.job_identifier}-pp"
+
+    @property
+    def postprocessing_job_name(self):
+        return f"projects/{self.gcp_project}/locations/{self.region}/jobs/{self.postprocessing_job_id}"
+
+    @property
+    def postprocessing_job_console_url(self):
+        return (
+            f"https://console.cloud.google.com/run/jobs/details/{self.region}"
+            f"/{self.postprocessing_job_id}/executions?project={self.gcp_project}"
+        )
+
+    # todo: aws-shared (see file comment)
```

Comment:
Agree that it would be good to consolidate this with the AWS build_image, but I do get that different things could be needed.

Reply:
Moved in #443.